Library/Homebrew/brew.sh

```diff
@@ -426,21 +426,14 @@ then
   HOMEBREW_SYSTEM="Macintosh"
   [[ "${HOMEBREW_PROCESSOR}" == "x86_64" ]] && HOMEBREW_PROCESSOR="Intel"
   HOMEBREW_MACOS_VERSION="$(/usr/bin/sw_vers -productVersion)"
+  HOMEBREW_OS_VERSION="macOS ${HOMEBREW_MACOS_VERSION}"
   # Don't change this from Mac OS X to match what macOS itself does in Safari on 10.12
   HOMEBREW_OS_USER_AGENT_VERSION="Mac OS X ${HOMEBREW_MACOS_VERSION}"
 
   # Intentionally set this variable by exploding another.
   # shellcheck disable=SC2086,SC2183
   printf -v HOMEBREW_MACOS_VERSION_NUMERIC "%02d%02d%02d" ${HOMEBREW_MACOS_VERSION//./ }
 
-  # Don't include minor versions for Big Sur and later.
```

Comment:
Why this change? Feels like we need to audit uses of `HOMEBREW_OS_VERSION` to check this won't have unintended effects.

Reply:
This change grew out of the discussion here: https://github.com/Homebrew/brew/pull/13664#discussion_r939689074

I can revert it, since it's not actually related to what I want to do here anymore (it was tangentially related to what I was trying to do in `brew.sh` earlier).

Reply:
Ah, gotcha. I'm fine with it if you're happy with the various usages being fine 👍🏻

Reply:
My understanding was this was done to fix our analytics not being able to handle it, but that's fixed now.

In any case, the previous code didn't strip enough on Big Sur, which still comes with patch versions these days (e.g. `11.5.1`) but Monterey does not (e.g. `12.5`), and that seemed to work ok (i.e. the rest of the code dealt with `macOS 11.5` just fine). This suggests to me that keeping the entire version number doesn't hurt.